pkg/flow/internal/controller/node_custom_component.go

```diff
@@ -0,0 +1,366 @@
+func (cn *CustomComponentNode) Exports() component.Exports {
+	cn.exportsMut.RLock()
+	defer cn.exportsMut.RUnlock()
+	return cn.exports
+}
+
+func (cn *CustomComponentNode) LastUpdateTime() time.Time {
+	return cn.lastUpdateTime.Load()
+}
+
+// setExports is called whenever the managed custom component updates. e must be the
+// same type as the registered exports type of the managed custom component.
+func (cn *CustomComponentNode) setExports(e component.Exports) {
+	// Some components may aggressively reexport values even though no exposed
+	// state has changed. This may be done for components which always supply
+	// exports whenever their arguments are evaluated without tracking internal
+	// state to see if anything actually changed.
+	//
+	// To avoid needlessly reevaluating components we'll ignore unchanged
+	// exports.
+	var changed bool
+
+	cn.exportsMut.Lock()
+	if !reflect.DeepEqual(cn.exports, e) {
+		changed = true
+		cn.exports = e
+	}
+	cn.exportsMut.Unlock()
+
+	if changed {
+		// Inform the controller that we have new exports.
+		cn.lastUpdateTime.Store(time.Now())
+		cn.OnNodeWithDependantsUpdate(cn)
+	}
+}
+
+// CurrentHealth returns the current health of the CustomComponentNode.
+//
+// The health of a CustomComponentNode is determined by combining:
+//
+//  1. Health from the call to Run().
+//  2. Health from the last call to Evaluate().
+//  3. Health reported from the custom component.
+func (cn *CustomComponentNode) CurrentHealth() component.Health {
+	cn.healthMut.RLock()
+	defer cn.healthMut.RUnlock()
+	return component.LeastHealthy(cn.runHealth, cn.evalHealth, cn.managed.CurrentHealth())
+}
+
+// TODO implement debugInfo?
+func (cn *CustomComponentNode) DebugInfo() interface{} {
+	cn.mut.RLock()
+	defer cn.mut.RUnlock()
+	return nil
+}
+
+// setEvalHealth sets the internal health from a call to Evaluate. See Health
+// for information on how overall health is calculated.
+func (cn *CustomComponentNode) setEvalHealth(t component.HealthType, msg string) {
+	cn.healthMut.Lock()
+	defer cn.healthMut.Unlock()
+
+	cn.evalHealth = component.Health{
+		Health:     t,
+		Message:    msg,
+		UpdateTime: time.Now(),
+	}
+}
+
+// setRunHealth sets the internal health from a call to Run. See Health for
+// information on how overall health is calculated.
+func (cn *CustomComponentNode) setRunHealth(t component.HealthType, msg string) {
+	cn.healthMut.Lock()
+	defer cn.healthMut.Unlock()
+
+	cn.runHealth = component.Health{
+		Health:     t,
+		Message:    msg,
+		UpdateTime: time.Now(),
+	}
+}
+
+// ModuleIDs returns the current list of custom components that this component is
+// managing.
+func (cn *CustomComponentNode) ModuleIDs() []string {
+	return cn.moduleController.ModuleIDs()
+}
+
+// BlockName returns the name of the block.
+func (cn *CustomComponentNode) BlockName() string {
+	return cn.componentName
+}
+
+// This node does not manage any component.
```

Comment:
Is that true here?

Reply:
I had a different definition of module in mind and I thought that this node was managing a module and not a component. I plan on following what Robert said above and to have this function return something